### ISSUE TITLE
Continued CORS Issues

### ISSUE THREAD

**Issue by pdfowler** [mosch/react-avatar-editor#268]
We have a very rudimentary CORS setup in our AWS Bucket, but cannot get anything to load into the editor. We receive the following error when trying to initialize the Avatar Editor with an image hosted in our AWS Bucket:

> Access to Image at 'https://ourbucket.s3.amazonaws.com/c15a53ff-9aae-4b94-b83f-23f3e96a00b6-profile-image-1537290670' from origin 'https://ourserver.ourcompany.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'https://ourserver.ourcompany.com' is therefore not allowed access.

Our AWS Config is rudimentary:

```
    <CORSRule>
        <AllowedOrigin>https://*.ourcompany.com</AllowedOrigin>
        <AllowedMethod>GET</AllowedMethod>
        <AllowedMethod>HEAD</AllowedMethod>
        <AllowedMethod>PUT</AllowedMethod>
        <ExposeHeader>Access-Control-Allow-Origin</ExposeHeader>
        <AllowedHeader>*</AllowedHeader>
        <MaxAgeSeconds>3600</MaxAgeSeconds>
    </CORSRule>`
```

And our component is configured with the `crossOrigin` prop set to `anonymous`.

In our local dev environment, with its own s3 bucket with the same config (with `localhost:*` as origin), it functions as expected and loads the image into the editor.

#### Notes:

1. We have focused on editing the S3 config, and have had occasional success on production, only to have it break again when trying on a different image.
1. The same image loads properly elsewhere in the app. When inspecting the request, the normal `GET`s don't set the `origin`, whereas the `origin` is "properly" set when the AvatarEditor requests the url.
1. Setting `crossOrigin` to the empty string loads the image, but runs into the "Tainted Canvas" problem discussed elsewhere
1. We have tried setting `crossOrigin` locally to `use-credentials` and it seems to work locally. Although the function of this is somewhat unclear. This will be the next change we roll to production to attempt to resolve the issue.

#### TLDR:

* CORS config appears correct - does anyone see anything glaringly wrong?
* Can someone better explain the `use-credentials` option for `crossOrigin`?

**Comment by mosch** [comment]
I'm closing this now. Thanks for this detailed request, however it looks like your question is more about AWS than this component.

**Comment by Munsterberg** [comment]
@pdfowler Did you end up finding a solution to this problem? We're experiencing something very similar.

**Comment by pdfowler** [comment]
Still haven't figured it out - every time it seems like we have it working, it works only intermittently then not at all. Using "anonymous" for `crossOrigin` ended up making things worse.

This is a bit of an edge case in our app so it hasn't drawn very much attention and we have some workarounds in place for our team. I'll update in the future if I ever work things out

**Comment by Munsterberg** [comment]
Thanks for the quick response! I will do the same if we find a solution. Best of luck :)

**Comment by rla** [comment]
This could be related to caching a non-CORS image request by browsers (normal `<img>` tag without crossorigin attribute). The behavior and result is described in <https://bugs.chromium.org/p/chromium/issues/detail?id=409090>

**Comment by timB73** [comment]
You can fix the CORS error from caching behaviour by adding dummy parameter like ?x-request=html in original requests for the image. Described here: https://serverfault.com/questions/856904/chrome-s3-cloudfront-no-access-control-allow-origin-header-on-initial-xhr-req/856948
